lib/FrontendTool/FrontendTool.cpp

```diff
@@ -1384,6 +1384,7 @@ static bool serializeSIB(SILModule *SM, const PrimarySpecificPaths &PSPs,
   serializationOpts.OutputPath = moduleOutputPath.c_str();
   serializationOpts.SerializeAllSIL = true;
   serializationOpts.IsSIB = true;
+  serializationOpts.SerializeOptionsForDebugging = true;
```

Comment:
I'll refine after https://github.com/apple/swift/pull/32237 merged. I'll serialize options only when `-lto=swift` is given.